modules/ip6/datapath/ndp_na_output.c

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2024 Robin Jarry
+
+#include <gr_control_input.h>
+#include <gr_graph.h>
+#include <gr_icmp6.h>
+#include <gr_ip6_control.h>
+#include <gr_ip6_datapath.h>
+#include <gr_log.h>
+#include <gr_mbuf.h>
+#include <gr_trace.h>
+
+#include <rte_byteorder.h>
+#include <rte_ether.h>
+#include <rte_graph_worker.h>
+#include <rte_ip6.h>
+
+enum {
+	OUTPUT = 0,
+	EDGE_COUNT,
+};
+
+static uint16_t ndp_na_output_process(
+	struct rte_graph *graph,
+	struct rte_node *node,
+	void **objs,
+	uint16_t nb_objs
+) {
+	const struct nexthop *local, *remote;
+	struct ip6_local_mbuf_data *d;
+	struct icmp6_neigh_advert *na;
+	struct icmp6_opt_lladdr *ll;
+	struct icmp6_opt *opt;
+	struct rte_mbuf *mbuf;
+	uint16_t payload_len;
+	struct icmp6 *icmp6;
+
+	for (uint16_t i = 0; i < nb_objs; i++) {
+		mbuf = objs[i];
+
+		local = ndp_na_output_mbuf_data(mbuf)->local;
+		remote = ndp_na_output_mbuf_data(mbuf)->remote;
+
+		rte_pktmbuf_reset(mbuf);
```

Comment:
I think I already commented on this but I don't remember the reason.
Why call rte_pktmbuf_reset()?

Here, rather than call rte_pktmbuf_reset(), you can simply drop any data by trimming (rte_pktmbuf_trim(m, rte_pktmbuf_pkt_len(m))).
This will preserve current data offset (and this reply packet will likely get symmetrically recapsulated on its output path).